packages/shared/decorators/with-loader.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from "react";
+
+const withLoader = <T extends { [k: string]: any }>(
```

Comment:
Because of
`type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;`
I think that extending an object will be enough.
```suggestion
const withLoader = <T extends {}>(
```